### PR TITLE
Migrate action runner image from data platform to new repository.

### PR DESCRIPTION
**Pull request by BrianEllwood**
Migrate action runner image from data platform to new repository.

- copied dockerfile and entrypoint.sh 
- amended code to reflect new repository
- created a dev container

This work is contributes to this  [ticket](https://github.com/ministryofjustice/data-platform/issues/4004)

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "analytical-platform-actions_runner",
```
Comment (jacobwoffenden) [inline .devcontainer/devcontainer.json:2]:
```suggestion
  "name": "analytical-platform-actions-runner",
```

File: Dockerfile [hunk 2]
```diff
@@ -1,44 +1,57 @@
...
+    && mkdir --parents ${CONTAINER_HOME} \
+    && chown --recursive ${CONTAINER_USER}:${CONTAINER_GROUP} ${CONTAINER_HOME}
+
+COPY --chown=${CONTAINER_USER}:${CONTAINER_GROUP} src/usr/local/bin/actions-runner-utils /usr/local/bin/actions-runner-utils
+
+RUN apt-get update \
+    && apt-get install --yes --no-install-recommends \
+      ca-certificates=20230311ubuntu0.22.04.1 \
+      #curl=7.81.0-1ubuntu1.15 \
+      curl=7.81.0-1ubuntu1.16 \
```
Comment (jacobwoffenden) [inline Dockerfile:39]:
```suggestion
      curl=7.81.0-1ubuntu1.16 \
```

File: README.md
Comment (jacobwoffenden) [inline README.md:1]: The README needs the template stuff cutting out, and references to `analytical-platform-image-build-template` need replacing

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (BrianEllwood) [reply .devcontainer/devcontainer.json]: changes comited

File: Dockerfile
Comment (jacobwoffenden) [inline Dockerfile:1]: There are things like `SHELL` (albeit it doesn't match https://github.com/ministryofjustice/analytical-platform-image-build-template/blob/main/Dockerfile#L17) that have been ported from the template, but not pinning the SHA, what is in scope for this?

File: README.md [hunk 3]
```diff
@@ -1,91 +1,12 @@
...
-- test
-  - container-structure-test.yml
-- Dockerfile
-- .editorconfig
-- LICENSE
-- .gitignore
-- README
-- Ministry of Justice Compliance Badge (public repositories only)
-```
+## Readme
```
Comment (jacobwoffenden) [inline README.md:9]: Not sure this heading needs to be in the README

File: README.md [hunk 4]
```diff
@@ -1,91 +1,12 @@
...
 
-## Setup Instructions
+The source repository had no README to pull accross
```
Comment (jacobwoffenden) [inline README.md:11]:
```suggestion
The source repository had no README to pull across
```

File: README.md
Comment (jacobwoffenden) [inline README.md:1]: References to `analytical-platform-image-build-template` have not been replaced

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (AntFMoJ) [reply Dockerfile:1]: @jacobwoffenden we haven't really changed the Dockerfile from what it was in the data-platform repo, should we have added changes present in the `analytical-platform-image-build-template` Dockerfile?

Comment (jacobwoffenden) [reply Dockerfile:1]: sure, I should've looked at the source Dockerfile first 😬

**Review by jacobwoffenden — APPROVED** [review]

**Pull request merged**